src/middleware/validation.ts

```diff
@@ -0,0 +1,53 @@
+import { NextFunction, Request, Response } from "express";
+import {
+  fullToDoValidator,
+  partialToDoValidator,
+  isValueOfType,
+} from "../models/validators";
+import { errors } from "../utils/errors";
+
+const sortByProperties = ["createdAt", "priority", "title", "text"];
+const orderProperties = ["asc", "desc"];
+
+export const validateToDo = {
+  fullValidation: async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      await fullToDoValidator.validateAsync(req.body);
+      return next();
+    } catch (e) {
+      return next(errors.invalidParameter("Invalid ToDo"));
+    }
+  },
+  partialValidation: async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      await partialToDoValidator.validateAsync(req.body);
+      return next();
+    } catch (e) {
+      return next(errors.invalidParameter("Invalid ToDo"));
```

Comment:
Please try to return more explanatory error messages, which is the first wrong field
Like "Missing or invalid parameter 'title'"